File: frontend/src/types/order.ts
export interface Beer {
    nameId: string;
    name: string;
    price: number;
    quantity?: number;
  }

  export interface OrderItemDetail {
    name: string;
    quantity: number;
    price_per_unit: number;
    total_price: number;
    subtotal: number;
    discounts: number;
  }
  
  export interface OrderAdd {
    nameId: string;
    name: string;
    quantity: number;
    price: number;

  }

  export interface OrderRound {
    items: OrderItemDetail[];
    created: string;
  }
  
  export interface Order {
    id: string;
    status: 'pending' | 'paid';
    subtotal: number;
    taxes: number;
    discounts: number;
    rounds: OrderRound[];
  }
  
  export interface Stock {
    last_updated: string;
    beers: Beer[];
  }
  

File: backend/src/domain/entities/order.py
from pydantic import BaseModel
from typing import List
from src.domain.entities.order_item import OrderItem
from src.domain.entities.round import Round

class Order(BaseModel):
    created: str
    paid: bool
    subtotal: float
    taxes: float
    discounts: float
    items: List[OrderItem]
    rounds: List[Round]